homeassistant/components/sensor/mqtt.py

```diff
@@ -167,19 +163,13 @@ def message_received(topic, payload, qos):
                 self._expiration_trigger = async_track_point_in_utc_time(
                     self.hass, self.value_is_expired, expiration_at)
 
-            if self._json_attributes:
-                self._attributes = {}
-                try:
-                    json_dict = json.loads(payload)
-                    if isinstance(json_dict, dict):
-                        attrs = {k: json_dict[k] for k in
-                                 self._json_attributes & json_dict.keys()}
-                        self._attributes = attrs
-                    else:
-                        _LOGGER.warning("JSON result was not a dictionary")
-                except ValueError:
-                    _LOGGER.warning("MQTT payload could not be parsed as JSON")
-                    _LOGGER.debug("Erroneous JSON: %s", payload)
+            try:
+                json_dict = json.loads(payload)
+                if isinstance(json_dict, dict):
+                    self._attributes = json_dict
```

Comment:
Please add an `attribute_template` to replace the `json_attributes`.
For example, Tasmota sends data from all connected sensors in a single message, we don't want all that to show up as attributes.

Also, this means the value picked as state will also be duplicated as an attribute?

Reply:
I think that a better solution will be to allow a different attribute topic.